File: src/utils/logger.ts
/**
 * Minimal leveled logger. info/warn go to stderr (so stdout stays clean for
 * piped JSON/data); error always goes to stderr. info/warn are gated by verbose.
 */

export interface Logger {
  info(message: string): void;
  warn(message: string): void;
  error(message: string): void;
}

export function createLogger(verbose: boolean): Logger {
  return {
    info(message: string): void {
      if (verbose) console.error(message);
    },
    warn(message: string): void {
      if (verbose) console.warn(message);
    },
    error(message: string): void {
      console.error(message);
    },
  };
}

File: apps/cli/src/util/logger.rs
/**
 * Minimal leveled logger. info/warn go to stderr (so stdout stays clean
 * for piped JSON/data); error always goes to stderr. info/warn are gated
 * by verbose.
 */

pub struct Logger {
    verbose: bool,
}

impl Logger {
    pub fn info(&self, msg: &str) {
        if self.verbose {
            eprintln!("{}", msg);
        }
    }
    pub fn warn(&self, msg: &str) {
        if self.verbose {
            eprintln!("{}", msg);
        }
    }
    pub fn error(&self, msg: &str) {
        eprintln!("{}", msg);
    }
}

pub fn create_logger(verbose: bool) -> Logger {
    Logger { verbose }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn logger_creates_with_verbose_flag() {
        let log = create_logger(true);
        assert!(log.verbose);
    }

    #[test]
    fn logger_creates_without_verbose() {
        let log = create_logger(false);
        assert!(!log.verbose);
    }
}
